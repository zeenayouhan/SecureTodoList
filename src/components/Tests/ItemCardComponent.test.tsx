import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import ItemCardComponent from '../ItemCard.component';

describe('ItemCardComponent', () => {
  it('renders correctly', () => {
    const onPressUpdate = jest.fn();
    const onPressRemove = jest.fn();
    const { getByText, getByTestId } = render(
      <ItemCardComponent
        label='Test Label'
        index='1'
        onPressUpdate={onPressUpdate}
        onPressRemove={onPressRemove}
      />
    );

    const label = getByText('Test Label');
    const removeButton = getByText('REMOVE');

    expect(label).toBeDefined();
    expect(removeButton).toBeDefined();

    fireEvent.press(removeButton);
    expect(onPressRemove).toHaveBeenCalledTimes(1);

    fireEvent.press(getByTestId('item-card'));
    expect(onPressUpdate).toHaveBeenCalledTimes(1);
  });
});

describe('Rendering', () => {
  it('should match to snapshot', () => {
    const onPressUpdate = jest.fn();
    const onPressRemove = jest.fn();
    const itemCardComponent = render(
      <ItemCardComponent
        label='Test Label'
        index='1'
        onPressUpdate={onPressUpdate}
        onPressRemove={onPressRemove}
      />
    );
    expect(itemCardComponent).toMatchSnapshot();
  });
});
