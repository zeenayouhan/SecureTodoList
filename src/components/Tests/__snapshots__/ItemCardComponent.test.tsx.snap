// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering should match to snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "backgroundColor": "#FFFFFF",
      "borderRadius": 60,
      "flexDirection": "row",
      "marginHorizontal": 40,
      "marginVertical": 20,
      "opacity": 1,
      "overflow": "hidden",
    }
  }
  testID="item-card"
>
  <View
    style={
      {
        "flexDirection": "row",
      }
    }
  >
    <View
      style={
        {
          "flexDirection": "row",
        }
      }
    >
      <Text
        style={
          {
            "color": "#182387",
            "fontSize": 120,
            "height": 200,
            "marginBottom": -80,
            "marginLeft": 40,
            "marginTop": -90,
          }
        }
      >
        â€¢
      </Text>
      <Text
        style={
          {
            "color": "#8E8E93",
            "fontSize": 18,
            "marginLeft": 10,
            "marginVertical": 40,
            "width": 320,
          }
        }
      >
        Test Label
      </Text>
    </View>
    <View
      style={
        {
          "alignContent": "flex-end",
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "marginRight": 40,
            "opacity": 1,
          }
        }
      >
        <Text
          style={
            {
              "color": "#8E8E93",
              "fontSize": 18,
              "marginLeft": 10,
              "marginVertical": 40,
              "width": 320,
            }
          }
        >
          REMOVE
        </Text>
      </View>
    </View>
  </View>
</View>
`;
